modules/consensus/accept_test.go

```diff
@@ -701,3 +701,64 @@ func TestTaxHardfork(t *testing.T) {
 		t.Fatal("siafund pool was not increased correctly")
 	}
 }
+
+// mockGateway implements modules.Gateway to mock the Broadcast method.
+type mockGateway struct {
+	modules.Gateway
+	numBroadcasts int
+}
+
+func (g *mockGateway) Broadcast(string, interface{}) {
+	g.numBroadcasts++
+	return
+}
+
+// TestAcceptBlockBroadcasts tests that AcceptBlock broadcasts valid blocks and
+// that managedAcceptBlock does not.
+func TestAcceptBlockBroadcasts(t *testing.T) {
+	cst, err := blankConsensusSetTester("TestAcceptBlockBroadcasts")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cst.Close()
+	mg := &mockGateway{Gateway: cst.cs.gateway}
+	cst.cs.gateway = mg
+
+	// Test that Broadcast is called for valid blocks.
+	b, _ := cst.miner.FindBlock()
+	err = cst.cs.AcceptBlock(b)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Sleep to wait for possible calls to Broadcast to complete. We cannot
+	// wait on a channel because we don't know how many times broadcast has
+	// been called.
+	time.Sleep(1)
```

Comment:
I can't come up with a better way to solve this other than Sleep. If you wait on a channel, you must wait on it x number of times, where x is the same number of times you expect `gateway.Broadcast` to be called. But then `g.channel <- struct{}{}` will block if Broadcast is called more times than the `<-g.channel` is called. The channel could be buffered instead, but then waiting on the channel would only block until the expected number of broadcasts, after which there could be additional broadcasts.